lib/jbuilder.rb

```diff
@@ -12,12 +12,18 @@ class Jbuilder
   @@ignore_nil    = false
   @@deep_format_keys = false
 
-  def initialize(options = {})
+  def initialize(options = nil)
     @attributes = {}
 
-    @key_formatter = options.fetch(:key_formatter){ @@key_formatter ? @@key_formatter.clone : nil}
-    @ignore_nil = options.fetch(:ignore_nil, @@ignore_nil)
-    @deep_format_keys = options.fetch(:deep_format_keys, @@deep_format_keys)
+    if options
+      @key_formatter = options[:key_formatter]
+      @ignore_nil = options[:ignore_nil]
```

Comment:
My ruby is bad now - will this be nil if it's unset which I know is falsy but..

Reply:
`options` can't be unset. It is a required parameter that defaults to `nil`.

I can change this to 

```ruby
    if options.nil?
      @key_formatter = @@key_formatter
      @ignore_nil = @@ignore_nil
      @deep_format_keys = @@deep_format_keys
    else
      @key_formatter = options[:key_formatter]
      @ignore_nil = options[:ignore_nil]
      @deep_format_keys = options[:deep_format_keys]
    end
```

to make it clearer (and perhaps safer).

Reply:
Yeah, `Hash`es will return `nil` if the key doesn't exist.

Moved it over to use key args, which actually speeds things up a bit more.